GitHubContentUtility/Helpers/JwtHelper.cs

```diff
@@ -0,0 +1,73 @@
+﻿// ------------------------------------------------------------------------------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using Jose;
+using Org.BouncyCastle.Crypto;
+using Org.BouncyCastle.Crypto.Parameters;
+using Org.BouncyCastle.Math;
+using Org.BouncyCastle.OpenSsl;
+using System.IO;
+using System.Security.Cryptography;
+
+namespace GitHubContentUtility.Helpers
+{
+    /// <summary>
+    /// Provides methods for extracting a private key and using it to encode a Jwt token.
+    /// </summary>
+    public class JwtHelper
+    {
+        /// <summary>
+        /// Gets private key and uses it to decode the jwt token.
+        /// </summary>
+        /// <param name="key"></param>
+        /// <param name="payload"></param>
+        /// <returns>jwt token</returns>
+        public static string CreateEncodedJwtToken(string key, Dictionary<string, object> payload)
+        {
+            // Generate JWT
+            using var rsa = new RSACryptoServiceProvider();
+            var rsaParams = ToRSAParameters(GetPrivateKey(key));
+            rsa.ImportParameters(rsaParams);
+            return JWT.Encode(payload, rsa, JwsAlgorithm.RS256);
+        }
+        private static RsaPrivateCrtKeyParameters GetPrivateKey(string privateKey)
+        {
+            using var privateKeyReader = new StringReader(privateKey);
+            var pemReader = new PemReader(privateKeyReader);
+            var keyPair = (AsymmetricCipherKeyPair)pemReader.ReadObject();
+            return (RsaPrivateCrtKeyParameters)keyPair.Private;
+        }
+
+        private static RSAParameters ToRSAParameters(RsaPrivateCrtKeyParameters privKey)
```

Comment:
Please share a link as a comment when the code is taken from elsewhere, it will be easier to review. That way, we can review the licensing implications of porting the code as well. The original code is available as a NuGet package: `BouncyCastle`. Please consider consuming that.

Reply:
Used `Portable.BouncyCastle` `1.8.6.7` to remove the concrete implementations.
Also added the source code reference at the top of the file.